src/SignalR/clients/ts/signalr/src/HttpConnection.ts

```diff
@@ -293,13 +293,13 @@ export class HttpConnection implements IConnection {
         }
 
         if (transportExceptions.length > 0) {
-            throw new Error(`Unable to connect to the server with any of the available transports. ${transportExceptions.join(" ")}`);
+            return Promise.reject(new Error(`Unable to connect to the server with any of the available transports. ${transportExceptions.join(" ")}`));
         }
-        throw new Error("None of the transports supported by the client are supported by the server.");
+        return Promise.reject(new Error("None of the transports supported by the client are supported by the server."));
     }
 
     private constructTransport(transport: HttpTransportType) {
-        switch (transport) {
+    switch (transport) {
             case HttpTransportType.WebSockets:
                 if (!this.options.WebSocket) {
                     throw new Error("'WebSocket' is not supported in your environment.");
```

Comment:
What about this throw?

Reply:
And the ones in `resolveTransport` and `resolveUrl`?

Reply:
Errors in constructTransport and resolveTransport are caught and aggregated into a rejected Promise. `resolveUrl` is only used in the constructor so that should be fine just throwing an error